testtools/deferredruntest.py

```diff
@@ -232,8 +233,10 @@ def _run_core(self):
             [error_observer.gotEvent, full_observer.emit],
             self._blocking_run_deferred, spinner)
 
+        full_log.seek(0, os.SEEK_SET)
+        log_content = full_log.readlines()
         self.case.addDetail(
-            'twisted-log', Content(UTF8_TEXT, full_log.readlines))
+            'twisted-log', Content(UTF8_TEXT, lambda: log_content))
```

Comment:
If you set the Content instance's get_bytes to full_log.readlines, then it gets to iterate over it once and then will be at EOF if it tries to do it again.  This means that you get confusing results if, for example, you try to use an Equals matcher on it: it fails the match, then tries to iterate over the content again to construct the failure message and you just get the empty string.

But Robert's suggestions of using full_log.getvalue() and text_content() are better still, and make this moot.